src/components/dialog-full-screen/dialog-full-screen.test.tsx

```diff
@@ -0,0 +1,432 @@
+  expect(closeButton).toHaveAttribute("data-element", "foo");
+  expect(closeButton).toHaveAttribute("data-role", "bar");
+});
+
+test("renders the element given in the `headerChildren` prop", () => {
+  render(
+    <DialogFullScreen
+      open
+      headerChildren={<a href="/foo">link as header child</a>}
+    />
+  );
+  expect(
+    screen.getByRole("link", { name: "link as header child" })
+  ).toBeInTheDocument();
+});
+
+// test here for coverage only - disableContentPadding prop already tested in both Playwright and Chromatic
+test("padding is removed from the content when the `disableContentPadding` prop is passed", () => {
+  render(
+    <DialogFullScreen open disableContentPadding>
+      <div>test content</div>
+    </DialogFullScreen>
+  );
+
+  expect(screen.getByTestId("dialog-full-screen-content")).toHaveStyleRule(
+    "padding",
+    "0"
+  );
+});
+
+/** Remove this when after Pages is re-written */
+// TODO: in for coverage only. Prop appears not to be used - to be removed (FE-6774)
+test("applies the appropriate styles when the `pagesStyling` prop is set", () => {
+  render(<DialogFullScreen open pagesStyling />);
+
+  const dialog = screen.getByRole("dialog");
+
+  expect(dialog).toHaveStyleRule("padding", "0", {
+    modifier: `${StyledContent}`,
+  });
+
+  expect(dialog).toHaveStyleRule("margin", "0", {
+    modifier: `${StyledIconButton}`,
+  });
+  expect(dialog).toHaveStyleRule("position", "absolute", {
+    modifier: `${StyledIconButton}`,
+  });
+  expect(dialog).toHaveStyleRule("right", "33px", {
+    modifier: `${StyledIconButton}`,
+  });
+  expect(dialog).toHaveStyleRule("top", "32px", {
+    modifier: `${StyledIconButton}`,
+  });
+  expect(dialog).toHaveStyleRule("z-index", "1", {
+    modifier: `${StyledIconButton}`,
+  });
+
+  expect(dialog).toHaveStyleRule("padding", "32px 32px 0", {
+    modifier: `${StyledFullScreenHeading}`,
+  });
+
+  expect(dialog).toHaveStyleRule("width", "auto", {
+    modifier: `${StyledHeading}`,
+  });
+  expect(dialog).toHaveStyleRule("padding-top", "4px", {
+    modifier: `${StyledHeading}`,
+  });
+
+  expect(dialog).toHaveStyleRule("width", "100%", {
+    modifier: `${StyledHeading} ${StyledHeader}`,
+  });
+  expect(dialog).toHaveStyleRule("box-sizing", "content-box", {
+    modifier: `${StyledHeading} ${StyledHeader}`,
+  });
+  expect(dialog).toHaveStyleRule("margin", "0 0 0 3px", {
+    modifier: `${StyledHeading} ${StyledHeader}`,
+  });
+});
+
+// these tests should be rewritten to use toHaveStyle, to avoid passing when they shouldn't. The first should be:
+// - toHaveStyle({ "overflow-y": "auto" })
+// and the second
+// - toHaveStyle({ "overflow-y": "inherit" })
+// These don't work at the moment due to :has not being implemented in jsdom before v23.2.0, which would require a
+// jest upgrade.
+// toHaveStyleRule is more forgiving as it passes when the rule is passed to the styled component even if it's
+// overridden by another rule on the same component
```

Comment:
**suggestion(non-blocking)**: how about raising a ticket for this and rewording the comment to make it more actionable? Here is a proposed suggestion: 

```suggestion
/* TODO: <ticket number> Update these tests to use toHaveStyle to avoid false positives:
   - toHaveStyle({ "overflow-y": "auto" })
   - toHaveStyle({ "overflow-y": "inherit" })
   This requires upgrading jsdom, as :has() is not supported in jsdom before v23.2.0.
   toHaveStyleRule is currently used, but it is less strict and passes even if styles are overridden. */
```

Additionally, [once Carbon supports Jest v29](https://github.com/Sage/carbon/pull/6916), we could consider upgrading JSDOM independently to Jest in the near future if it doesn't cause issues with Jest.